web/src/user_profile.js

```diff
@@ -278,6 +285,10 @@ export function show_user_profile(user, default_tab_key = "profile-tab") {
         !user.is_system_bot &&
         people.is_person_active(user.user_id);
     const groups_of_user = user_groups.get_user_groups_of_user(user.user_id);
+    const can_manage_profile =
+        people.is_person_active(user.user_id) &&
+        page_params.is_admin &&
+        !people.is_current_user(user.email);
```

Comment:
We should never do checks with `email` when we have a user ID; you want `people.is_my_user_id`

Reply:
Changes this, and also rewrote the check for whether to show it to be more readable and include deactivated users; I tested and the modal seems to work fine for editing them (and in fact that's a useful feature at times):

``` diff
diff --git a/web/src/user_profile.js b/web/src/user_profile.js
index 360c3bbf27..95fb4d3b0a 100644
--- a/web/src/user_profile.js
+++ b/web/src/user_profile.js
@@ -292,10 +292,11 @@ export function show_user_profile(user, default_tab_key = "profile-tab") {
         !user.is_system_bot &&
         people.is_person_active(user.user_id);
     const groups_of_user = user_groups.get_user_groups_of_user(user.user_id);
+    // We currently have the main UI for editing your own profile in
+    // settings, so can_manage_profile is artifically false for those.
     const can_manage_profile =
-        ((people.is_person_active(user.user_id) && page_params.is_admin) ||
-            (user.is_bot && user.bot_owner_id && user.bot_owner_id === page_params.user_id)) &&
-        user.is_system_bot !== true &&
+        (people.can_admin_user(user) || page_params.is_admin) &&
+        !user.is_system_bot &&
         !people.is_my_user_id(user.user_id);
     const args = {
         user_id: user.user_id,
```

Reply:
Opened https://github.com/zulip/zulip/issues/26693 for some follow-ups for this.